python/fledge/services/core/api/north.py

```diff
@@ -71,27 +76,51 @@ async def _get_north_schedules(storage_client):
         return []
 
     schedules = []
+    north_sch_dict = {}
     schedule_list = await server.Server.scheduler.get_schedules()
     latest_tasks = await _get_tasks_status()
+    try:
+        services_from_registry = ServiceRegistry.get(s_type="Northbound")
+    except DoesNotExist:
+        services_from_registry = []
     for sch in schedule_list:
         if sch.name in north_schedules:
-            task = latest_tasks.get(sch.name, None)
-            schedules.append({
-                'id': str(sch.schedule_id),
-                'name': sch.name,
-                'processName': sch.process_name,
-                'repeat': sch.repeat.total_seconds() if sch.repeat else 0,
-                'day': sch.day,
-                'enabled': sch.enabled,
-                'exclusive': sch.exclusive,
-                'taskStatus': None if task is None else {
-                    'state': [t.name.capitalize() for t in list(Task.State)][int(task['state']) - 1],
-                    'startTime': str(task['start_time']),
-                    'endTime': str(task['end_time']),
-                    'exitCode': task['exit_code'],
-                    'reason': task['reason'],
+            if sch.process_name != "north_C" and sch.schedule_type != 1:
+                task = latest_tasks.get(sch.name, None)
+                north_sch_dict = {
+                    'id': str(sch.schedule_id),
+                    'name': sch.name,
+                    'processName': sch.process_name,
+                    'repeat': sch.repeat.total_seconds() if sch.repeat else 0,
+                    'day': sch.day,
+                    'enabled': sch.enabled,
+                    'exclusive': sch.exclusive,
+                    'execution': 'task',
+                    'taskStatus': None if task is None else {
+                        'state': [t.name.capitalize() for t in list(Task.State)][int(task['state']) - 1],
+                        'startTime': str(task['start_time']),
+                        'endTime': str(task['end_time']),
+                        'exitCode': task['exit_code'],
+                        'reason': task['reason']
+                    }
                 }
-            })
+            else:
+                for s_record in services_from_registry:
+                    if s_record._name == sch.name:
+                        north_sch_dict = {
+                            'id': str(sch.schedule_id),
+                            'name': sch.name,
+                            'processName': sch.process_name,
+                            'enabled': sch.enabled,
+                            'execution': 'service',
```

Comment:
I would suggest to keep it type or executionType

Reply:
This is as per Mark's comment on JIRA
> my recommendation is to have have a key of “execution” with values of “task” or “service”